app/Helpers/MSGraphAPI/UserByApp.php

```diff
@@ -0,0 +1,216 @@
+                    break;
+                default:
+                    $search = $searchProperty && $searchValue ? '&$search="'. $searchProperty . ':' . $searchValue . '"' : null;
+                    break;
+            }
+            // if search then append to query
+            //$search = $searchProperty && $searchValue ? '&$search="'. $searchProperty . ':' . $searchValue . '"' : null;
+            //$search= $searchProperty && $searchValue ? '&$filter='. $searchProperty . ' eq \'' . $searchValue . '\'' : null;
+            $endpoint = '/users';
+            $url = $endpoint . '?$count=true&$orderBy=displayName' . $topN . $search . ' &$select=id,displayName,surname,givenName,userPrincipalName,mail,businessPhones,onPremisesSamAccountName,employeeId,' . $this->descriptionProperty . ',' . $this->employeeNumberProperty  . ',' . $this->mailProperty ;
+            # https://graph.microsoft.com/v1.0/users?$search="displayName:MyName"&$count=true&$orderBy=displayName&$select=id,displayName,surname,givenName,userPrincipalName,mail,businessPhones,onPremisesSamAccountName,employeeId,extension_e1fd504a055643d5b9d370b8b24b45ee_description,extension_e1fd504a055643d5b9d370b8b24b45ee_employeeNumber,extension_e1fd504a055643d5b9d370b8b24b45ee_mail
+            $request = $graph->createRequest("GET", $url);
+            $request->addHeaders(["ConsistencyLevel"=> "eventual"]);
+            $data=$request->execute();
+            $content = $data->getBody();
+            $users = collect(json_decode(json_encode($content['value']), FALSE));
+            if(empty($users) || count($users) >= 50 ) { 
+                return null; 
+            }
+            // if nextLink then loop
+            while(array_key_exists('@odata.nextLink',$content)){
+                $url = $content['@odata.nextLink'];
+                $request = $graph->createRequest("GET", $url);
+                $request->addHeaders(["ConsistencyLevel"=> "eventual"]);
+                $data=$request->execute();
+                $content = $data->getBody();
+                $newusers = collect(json_decode(json_encode($content['value']), FALSE));
+                $users = $users->merge($newusers);
+            }
+            
+            $cleanupUsers = $users->map(function ($user) use($checkAdmin){
+                if($checkAdmin){
+                    $isAdmin = $this->getUserRoleAssignments($user->id);
+                    $user->mensa_admin = $isAdmin ?? False;
+                    
+                }
+                $user->name = $user->displayName;
+                $user->email = $user->{$this->mailProperty} ?? null;
+                $user->phonenumber = $user->businessPhones[0] ?? null;
+                $user->description = isset($user->{$this->descriptionProperty}) ? Arr::join( $user->{$this->descriptionProperty} , ',' ) : null;
+                $user->lidnummer = $user->{$this->employeeNumberProperty} ?? null;
```

Comment:
Alles engels maar lidnummer niet

Reply:
ik blame @SubSide

Reply:
Wat zou er gebeuren als je die gewoon refactored 👀 
Je PR is al mega dus wat maakt het nu nog uit, xD